src/main/java/games/strategy/triplea/TripleAUnit.java

```diff
@@ -401,14 +400,15 @@ public static int getProductionPotentialOfTerritory(final Collection<Unit> units
 
   public static Unit getBiggestProducer(final Collection<Unit> units, final Territory producer, final PlayerID player,
       final GameData data, final boolean accountForDamage) {
-    final CompositeMatchAnd<Unit> factoryMatch = new CompositeMatchAnd<>(
-        Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player), Matches.unitIsBeingTransported().invert());
+    final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.<Unit>newCompositeBuilder()
```

Comment:
My vote would be for either the stream approach with add() calls lined up or `Match#newCompositeBuilder(Match<T>...)` to allow passing them to the builder. Either I think feels much cleaner than the existing approach and I vote strongly against the last option of passing the class.